products/jbrowse-cli/src/commands/text-index.ts

```diff
@@ -0,0 +1,398 @@
+        .get(urlIn, response => {
+          const parseStream = this.parseGff3Stream(response)
+          resolve(parseStream)
+        })
+        .on('error', (e: NodeJS.ErrnoException) => {
+          reject('fail')
+          if (e.code === 'ENOTFOUND') {
+            this.error('Bad file url')
+          } else {
+            this.error('Other error: ', e)
+          }
+        })
+    })
+  }
+
+  // Grab the remote file from urlIn, then unzip it before piping into
+  // parseGff3Stream for parsing.  Returns a promise for the resulting
+  // @gmod/gff stream.
+  handleGff3UrlWithGz(urlIn: string) {
+    const unzip = createGunzip()
+    const newUrl = new URL(urlIn)
+
+    return new Promise<Readable>((resolve, reject) => {
+      ;(newUrl.protocol === 'https:' ? httpsFR : httpFR)
+        .get(urlIn, response => {
+          resolve(this.parseGff3Stream(response.pipe(unzip)))
+        })
+        .on('error', (e: NodeJS.ErrnoException) => {
+          reject('fail')
+          if (e.code === 'ENOTFOUND') {
+            this.error('Bad file url')
+          } else {
+            this.error('Other error: ', e)
+          }
+        })
+    })
+  }
+
+  // Checks if the passed in string is a valid URL.
+  // Returns a boolean.
+  isURL(FileName: string) {
+    let url
+
+    try {
+      url = new URL(FileName)
+    } catch (_) {
+      return false
+    }
+
+    return url.protocol === 'http:' || url.protocol === 'https:'
+  }
+
+  // Function that takes in a gff3 readstream and parses through it and
+  // retrieves the needed attributes and information. Returns a @gmod/gff
+  // stream.
+  parseGff3Stream(gff3In: Readable): Readable {
+    return gff3In.pipe(gff.parseStream({ parseSequences: false }))
+  }
+
+  // Recursively goes through every record in the gff3 file and gets the
+  // desired attributes in the form of a JSON object. It is then pushed to
+  // gff3Stream in proper indexing format.
+  //
+  recurseFeatures(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    record: any,
+    gff3Stream: Readable,
+    attributes: string[],
+    trackID: string,
+  ) {
+    // goes through the attributes array and checks if the record contains the
+    // attribute that the user wants to search by. If it contains it, it adds
+    // it to the record object and attributes string
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const getAndPushRecord = (subRecord: any) => {
+      if (!subRecord) {
+        return
+      }
+      const locStr = `${subRecord['seq_id']};${subRecord['start']}..${subRecord['end']}`
+      const RecordValues: Array<string> = []
+      const RecordAttributes: Array<string> = []
+
+      RecordAttributes.push('locstring', 'TrackID')
+      RecordValues.push(locStr, trackID)
+
+      for (const x of attributes) {
+        RecordAttributes.push(x)
+      }
+
+      fs.writeFileSync(
+        path.join(__dirname, '..', '..', 'trix', 'meta.json'),
```

Comment:
just as a matter of note, the way that I test the program is to for example run `bin/run text-index --out ../../test_data/volvox/` and this will output to the volvox data directory, which I can then test in the browser (and then, I'd expect the files would be outputted to ../../test_data/volvox/trix/meta.json, etc.)

Reply:
I think this is a good idea. @TiaSFlores